numpy/core/src/multiarray/flagsobject.c

```diff
@@ -398,6 +420,35 @@ arrayflags_writeable_set(PyArrayFlagsObject *self, PyObject *obj)
     return 0;
 }
 
+static int
+arrayflags_warn_on_write_set(PyArrayFlagsObject *self, PyObject *obj)
+{
+    /*
+     * This code should go away in a future release, so do not mangle the
+     * array_setflags function with an extra kwarg
+     */
+    if (obj == NULL) {
+        PyErr_SetString(PyExc_AttributeError,
+                "Cannot delete flags _warn_on_write attribute");
+        return -1;
+    }
+    if (PyObject_IsTrue(obj)) {
```

Comment:
PyObject_IsTrue can return `-1`, you need to check for that

Reply:
fixing